src/trait-bounds.md

```diff
@@ -0,0 +1,131 @@
+# Trait and lifetime bounds
+
+> **<sup>Syntax</sup>**  
+> _TypeParamBounds_ :  
+> &nbsp;&nbsp; _TypeParamBound_ ( `+` _TypeParamBound_ )<sup>\*</sup> `+`<sup>?</sup>  
```

Comment:
Not actionable, but I wish we had syntax for this pattern that pops up everywhere in our grammar.

Reply:
Some of the Rust parsers that people have written define something for this ([example](https://github.com/nikomatsakis/rustypop/blob/6f94d89b6b032f4a75eb809a821cbf6bbe11102b/src/rusty.lalrpop#L1120)), but I'm not sure how far we want to stray from common ways of defining syntax.